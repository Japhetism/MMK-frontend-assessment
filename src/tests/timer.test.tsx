import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import assert from 'assert';
import Timer from '../components/timer';

describe('The component rendered', () => {

  it('renders the Timer component without crashing', () => {
    const durationOptions = [1,2,3,4,5];
    const setDisableUserInput = () => {};
    const setShowResult = () => {};
    const wrapper = shallow(<Timer
        durationOptions={durationOptions}
        setDisableUserInput={setDisableUserInput}
        setShowResult={setShowResult}
    />);
    expect(wrapper).toMatchSnapshot();
  });
})