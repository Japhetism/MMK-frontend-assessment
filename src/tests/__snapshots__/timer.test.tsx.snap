// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component rendered renders the Timer component without crashing 1`] = `
<styled.div>
  <CountdownTimer
    minutes={0}
    seconds={0}
  />
  <DurationSelector
    canStartTimer={false}
    durationOptions={
      Array [
        1,
        2,
        3,
        4,
        5,
      ]
    }
    minutes={0}
    setCanStartTimer={[Function]}
    setMinutes={[Function]}
    timerAction="Start"
  />
</styled.div>
`;
